Marlin/ultralcd.cpp

```diff
@@ -1285,6 +1279,18 @@ void kill_screen(const char* lcd_msg) {
       }
     }
 
+    inline void lcd_wait_for_homing() {
+      #if ENABLED(MANUAL_BED_LEVELING)
+        if (axis_homed[X_AXIS] && axis_homed[Y_AXIS] && axis_homed[Z_AXIS])
+          lcd_goto_screen(_lcd_level_bed_homing_done);
+      #else
+        if (no_reentrance) return;
+        while (!axis_homed[X_AXIS] || !axis_homed[Y_AXIS] || !axis_homed[Z_AXIS]) idle();
+        no_reentrance = true;
```

Comment:
Yes, I tested and it works both for manual bed leveling which I'm using right now and auto bed leveling (with no sensor just using z-switch, but it looks like has a shift in XY coordinates which is probably related to the offsets config).

Still waiting for my sensor, so will set it up as soon as it arrives, but it looks like will take some time.